spec/dummy/bin/fashionista.pl

```diff
@@ -0,0 +1,478 @@
+  
+  perl fashionista.pl --conf FILE 
+  # Specify the configuration file location
+ 
+  perl fashionista.pl --location URL 
+  # Specify the styleguide URL
+ 
+  perl fashionista.pl --help 
+  # Show this message (works also with --usage) 
+ 
+  Note: 
+   all command line options override their 
+   configuration file directives couterparts 
+USAGE
+    return 1;
+}
+my $conf_file = '../fashionista.conf';
+my ( $loglevel, $location, $usage );
+
+GetOptions(
+    'conf=s'     => \$conf_file,
+    'loglevel=s' => \$loglevel,
+    'location=s' => \$location,
+    'usage'      => \$usage,
+    'help'       => \$usage,
+) or usage and exit;
+
+if ($usage) { usage and exit; }
+
+# Conf init
+my ( %conf, %etc_conf );
+%conf = (
+    'dir'      => qq($Bin/../../../build),
+    'index'    => 'main.html',
+    'base_url' => 'http://localhost:3000/',
+    'log'      => {
+        log_level => 'INFO',
+        file      => {
+            filename  => qq($Bin/../fashionista.log),
+            pattern   => '[%d] [%p] %m%n',
+            mode      => 'append',
+            autoflush => '1',
+        },
+        screen => { stderr => 1, utf8 => 1, },
+    },
+);
+
+# Overload built-in configuration directives
+# if a configuration file is found
+if ( -e $conf_file ) {
+    %etc_conf = Config::General->new(
+        -ConfigFile            => $conf_file,
+        -MergeDuplicateBlocks  => 1,
+        -MergeDuplicateOptions => 1,
+        -IncludeGlob           => 1,
+        -UseApacheInclude      => 1,
+        -UTF8                  => 1,
+
+    )->getall();
+    for my $etc_conf_key ( keys %etc_conf ) {
+        $conf{$etc_conf_key} = $etc_conf{$etc_conf_key};
+    }
+}
+if ($loglevel) { $conf{'log'}{'log_level'} = $loglevel; }
+if ($location) { $conf{'log'}{'base_url'}  = $location; }
+
+# Log4perl init
+my $tt          = Template->new();
+my $l4p_conf    = q();
+my $tt_log_conf = <<'LOGCONF';
+log4perl.rootLogger=[% log.log_level %],file,screen
+log4perl.appender.file=Log::Log4perl::Appender::File
+log4perl.appender.file.create_at_logtime = 1
+log4perl.appender.file.layout=PatternLayout
+log4perl.appender.file.layout.ConversionPattern=[% log.file.pattern %]
+log4perl.appender.file.filename=[% log.file.filename %] 
+log4perl.appender.file.mode=[% log.file.mode %]
+log4perl.appender.file.autoflush=[% log.file.autoflush %]
+log4perl.appender.screen=Log::Log4perl::Appender::Screen
+log4perl.appender.screen.layout=Log::Log4perl::Layout::SimpleLayout
+log4perl.appender.screen.layout.ConversionPattern=[% log.screen.pattern %]
+LOGCONF
+
+$tt->process( \$tt_log_conf, \%conf, \$l4p_conf ) or croak $tt->error();
+if ( !-e $conf{'log'}{'file'}{'filename'} ) {
+    carp( sprintf 'Log file not found, creating : %s',
+        $conf{'log'}{'file'}{'filename'} );
+    touch( $conf{'log'}{'file'}{'filename'} );
+}
+Log::Log4perl->init( \$l4p_conf );
+my $log = Log::Log4perl->get_logger(__PACKAGE__);
+$log->info('Fashionista -  conf loaded');
```

Comment:
Can you elaborate ? Je ne vois pas d'erreur. Il y a un espace avant et un après, c'est voulu si tu parles de cela.